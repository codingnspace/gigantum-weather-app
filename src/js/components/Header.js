import React, {Component} from 'react';
import logoImg from 'images/bicycle.svg';

export default class Header extends Component {
  displayName: 'Header';

  render () {
    const {title, subtitle} = this.props;

    return (
      <div className='app-header relative'>
        <h1 className='app-title text-center'>{title}</h1>
        <h3 className='app-subtitle text-center'>{subtitle}</h3>
        <img className="app-logo" src={logoImg} alt="app-logo" />
      </div>
    );
  }
}
