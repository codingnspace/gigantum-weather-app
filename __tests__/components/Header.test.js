import renderer from 'react-test-renderer';
import Header from 'js/components/Header';
import React from 'react';

/**
* Snapshot testing can be very useful if used correctly, this example just shows how to use
* it but this is not the most useful snapshot, sometimes it is better to call it multiple
* times inside a test and take snapshots of the component changing throughout some interactions,
* Ex. snapshot default state, snapshot after click/mouseover, snapshot after mouseleave
*/
describe('Header snapshot Tests', () => {

  test('should render correctly', () => {
    const header = renderer.create(
      <Header/>
    );
    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
