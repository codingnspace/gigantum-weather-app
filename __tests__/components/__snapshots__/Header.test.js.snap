// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot Tests Sample of snapshot testing, not very useful, but shows how to do it 1`] = `
<div
  className="app-header relative"
>
  <img
    alt="app-logo"
    className="app-logo-left"
    src={Object {}}
  />
  <h1
    className="app-title text-center"
  />
  <h3
    className="app-subtitle text-center"
  />
  <img
    alt="app-logo"
    className="app-logo-right"
    src={Object {}}
  />
</div>
`;

exports[`Header snapshot Tests should render correctly 1`] = `
<div
  className="app-header relative"
>
  <img
    alt="app-logo"
    className="app-logo-left"
    src={Object {}}
  />
  <h1
    className="app-title text-center"
  />
  <h3
    className="app-subtitle text-center"
  />
  <img
    alt="app-logo"
    className="app-logo-right"
    src={Object {}}
  />
</div>
`;
